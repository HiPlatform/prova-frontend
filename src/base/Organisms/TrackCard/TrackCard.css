.trackCard {
  display: flex;
  flex-direction: row;
  margin: 6px 4px;
  align-items: stretch;
  border: 1px solid #d9dadb;
  border-radius: 5px;
  box-shadow: 0px 1px 0px 0px #d9dadb;
  flex-grow: 1; }
  .trackCard > img {
    width: 80px !important;
    height: 120px; }
  .trackCard__info {
    width: 100%;
    margin-left: 6px;
    display: flex;
    flex-direction: column; }
    .trackCard__info__title > .attributeField, .trackCard__info__extra > .attributeField {
      margin: 4px 0;
      padding-right: 4px;
      flex: 1 auto; }
    .trackCard__info__title {
      font-size: 13px;
      font-weight: bold; }
    .trackCard__info__extra {
      display: flex;
      flex-direction: row;
      align-items: stretch;
      font-size: 11px; }
