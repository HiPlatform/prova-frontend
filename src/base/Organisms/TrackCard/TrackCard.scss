@import "../../../variables";

.trackCard {
  display: flex;
  flex-direction: row;
  margin: 6px 4px;
  align-items: stretch;
  background-color: $cleanBackground;
  border-right: 1px solid $borderColor;
  border-radius: 5px;
  box-shadow: $shadow;
  flex-grow: 1;
  width: 100%;

  & > img {
    width: 80px !important;
    height: 120px;
  }

  &__info {
    width: 100%;
    margin-left: 6px;
    display: flex;
    flex-direction: column;

    &__title > .attributeField, &__extra > .attributeField {
      margin: 4px 0;
      padding-right: 4px;
      flex: 1 auto;
    }

    &__title {
      font-size: 13px;
      font-weight: bold;
    }

    &__extra {
      display: flex;
      flex-direction: row;
      align-items: stretch;
      font-size: 11px;
    }
  }
}