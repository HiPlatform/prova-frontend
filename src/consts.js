export const URLS = {
  SEARCH: 'https://api.spotify.com/v1/search',
};

export const CREDENTIALS = {
  headers: {
    Authorization: 'Bearer BQDGGAag6Hocue0iXDFWp0O9CdmjCSq6nI9R2-HE-_N-skmHG8V3btyKOZnilxWjn7cpkFMAbM70G8CH_-Y',
  },
};

export const HOMEPAGE_ACTIONS = {
  SEARCH_DATA: 'SEARCH_DATA',
  SEARCH_TYPE: 'SEARCH_TYPE',
  SEARCH_CHANGE: 'SEARCH_CHANGE',
  SEARCH_DETAILS: 'SEARCH_DETAILS',
  NAVIGATE: 'NAVIGATE',
};
