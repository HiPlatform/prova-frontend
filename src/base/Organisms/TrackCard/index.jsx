/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';

import { imageUrl, getArtistsName, getDuration } from '../../../helpers/cardsFunctions';

import Image from '../../Atoms/Image';
import AttributeField from '../../Molecules/AttributeField';

import './TrackCard.css';

const TrackCard = ({
  name,
  artists,
  album,
  duration_ms,
}) => (
  <div className="trackCard">
    <Image url={imageUrl(album, ['images'])} alt="album" />
    <div className="trackCard__info">
      <div className="trackCard__info__title">
        <AttributeField text={name} />
      </div>
      <div className="trackCard__info__extra">
        <AttributeField label="Artists" text={getArtistsName(artists).join(', ')} />
        <AttributeField label="Album Name" text={album.name} />
        <AttributeField label="Duration" text={getDuration(duration_ms)} />
      </div>
    </div>
  </div>
);

TrackCard.propTypes = {
  name: PropTypes.string.isRequired,
  artists: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  album: PropTypes.shape({
    name: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.shape({
      url: PropTypes.string.isRequired,
    })),
  }).isRequired,
  duration_ms: PropTypes.number.isRequired,
};

export default TrackCard;
