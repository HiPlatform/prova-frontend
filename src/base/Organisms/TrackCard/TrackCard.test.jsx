import React from 'react';
import { shallow } from 'enzyme';
import TrackCard from './index';
import { DEFAULT_TEXTS } from '../../../consts';

describe('TrackCard', () => {
  const props = {
    name: 'Some Track',
    artists: [{ name: 'Artist #1' }, { name: 'Artist #2' }],
    album: {
      name: 'Some album',
      images: [{ url: 'https://some.image/url' }]
    },
    duration_ms: 75 * 1000,
  };
  const component = shallow(<TrackCard {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should put the first image url in card header', () => {
    expect(component.find('Image').dive().prop('src')).toBe(props.album.images[0].url);
  });

  it('should render the album name given in the props', () => {
    const albumElement = component.find('.trackCard__info__extra')
      .find({ label: 'Album Name' })
      .render()
      .find('p');

    expect(albumElement.text()).toBe(props.album.name);
  });

  it('should render the duration in "minutes:seconds" based in the miliseconds length of the track', () => {
    const durationElement = component.find('.trackCard__info__extra')
      .find({ label: 'Duration' })
      .render()
      .find('p');

    expect(durationElement.text()).toBe('1:15');
  });

  it('should put a mock image url in card header if prop images has no images at all', () => {
    const album = {
      images: [],
    };
    component.setProps({ album });

    expect(component.find('Image').dive().prop('src')).toBe(DEFAULT_TEXTS.IMAGE);
  });

  it('should render the track name given in the props', () => {
    expect(component.find('.trackCard__info__title').render().text()).toBe(props.name);
  });

  it('should render all artists names', () => {
    const artistsElement = component.find('.trackCard__info__extra')
      .find({ label: 'Artists' })
      .render()
      .find('p');
    const names = props.artists.map(item => item.name);

    expect(artistsElement.text()).toBe(names.join(', '));
  });

  it('should render nothing when the artists prop is an empty array', () => {
    const artists = [];
    component.setProps({ artists });
    const artistsElement = component.find('.trackCard__info__extra')
      .find({ label: 'Artists' })
      .render()
      .find('p');

    expect(artistsElement.text()).toBe('');
  });
});
