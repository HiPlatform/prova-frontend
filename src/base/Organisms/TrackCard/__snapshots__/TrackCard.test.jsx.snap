// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackCard should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrackCard
    album={
        Object {
            "images": Array [
              Object {
                "url": "https://some.image/url",
              },
            ],
            "name": "Some album",
          }
    }
    artists={
        Array [
            Object {
              "name": "Artist #1",
            },
            Object {
              "name": "Artist #2",
            },
          ]
    }
    duration_ms={75000}
    name="Some Track"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt="album"
          url="https://some.image/url"
/>,
        <div
          className="trackCard__info"
>
          <div
                    className="trackCard__info__title"
          >
                    <AttributeField
                              label={null}
                              text="Some Track"
                    />
          </div>
          <div
                    className="trackCard__info__extra"
          >
                    <AttributeField
                              label="Artists"
                              text="Artist #1, Artist #2"
                    />
                    <AttributeField
                              label="Album Name"
                              text="Some album"
                    />
                    <AttributeField
                              label="Duration"
                              text="1:15"
                    />
          </div>
</div>,
      ],
      "className": "trackCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "album",
          "url": "https://some.image/url",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="trackCard__info__title"
>
              <AttributeField
                            label={null}
                            text="Some Track"
              />
</div>,
            <div
              className="trackCard__info__extra"
>
              <AttributeField
                            label="Artists"
                            text="Artist #1, Artist #2"
              />
              <AttributeField
                            label="Album Name"
                            text="Some album"
              />
              <AttributeField
                            label="Duration"
                            text="1:15"
              />
</div>,
          ],
          "className": "trackCard__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AttributeField
                label={null}
                text="Some Track"
/>,
              "className": "trackCard__info__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": null,
                "text": "Some Track",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <AttributeField
                  label="Artists"
                  text="Artist #1, Artist #2"
/>,
                <AttributeField
                  label="Album Name"
                  text="Some album"
/>,
                <AttributeField
                  label="Duration"
                  text="1:15"
/>,
              ],
              "className": "trackCard__info__extra",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Artists",
                  "text": "Artist #1, Artist #2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Album Name",
                  "text": "Some album",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Duration",
                  "text": "1:15",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt="album"
            url="https://some.image/url"
/>,
          <div
            className="trackCard__info"
>
            <div
                        className="trackCard__info__title"
            >
                        <AttributeField
                                    label={null}
                                    text="Some Track"
                        />
            </div>
            <div
                        className="trackCard__info__extra"
            >
                        <AttributeField
                                    label="Artists"
                                    text="Artist #1, Artist #2"
                        />
                        <AttributeField
                                    label="Album Name"
                                    text="Some album"
                        />
                        <AttributeField
                                    label="Duration"
                                    text="1:15"
                        />
            </div>
</div>,
        ],
        "className": "trackCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "album",
            "url": "https://some.image/url",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="trackCard__info__title"
>
                <AttributeField
                                label={null}
                                text="Some Track"
                />
</div>,
              <div
                className="trackCard__info__extra"
>
                <AttributeField
                                label="Artists"
                                text="Artist #1, Artist #2"
                />
                <AttributeField
                                label="Album Name"
                                text="Some album"
                />
                <AttributeField
                                label="Duration"
                                text="1:15"
                />
</div>,
            ],
            "className": "trackCard__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AttributeField
                  label={null}
                  text="Some Track"
/>,
                "className": "trackCard__info__title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": null,
                  "text": "Some Track",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <AttributeField
                    label="Artists"
                    text="Artist #1, Artist #2"
/>,
                  <AttributeField
                    label="Album Name"
                    text="Some album"
/>,
                  <AttributeField
                    label="Duration"
                    text="1:15"
/>,
                ],
                "className": "trackCard__info__extra",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Artists",
                    "text": "Artist #1, Artist #2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Album Name",
                    "text": "Some album",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Duration",
                    "text": "1:15",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
